### PR TITLE
Full Site Editing: Blank Theme, child of Twenty Nineteen

### LINKED ISSUE #33160: Full Site Editing: create Twenty Nineteen child theme [3]
The purpose of this task is to convert our existing "Blank" theme prototype to Twenty Nineteen child theme in order to deal with the complexities of the template hierarchy in a scoped-down way. For now, we will just take over page rendering to account for the new FSE templates, while the rest should be handled by the parent theme.

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request

* Create a new Blank Theme, child of Twenty Nineteen, that handles how pages with Full Site Editing templates are rendered.

Notes on using Twenty Nineteen styles:
- The default alignment is awkwardly off-centered to the left which causes the template to look weird.
A quick workaround would be assigning a `wide` (which is centered) or `full` alignment to the template part and content slot blocks.
- The main headings have a short line above as a `:before` pseudo element, which is not applied to the site title: `h1:not(.site-title):before, h2:before { ... }`.
I haven't overridden it in the child style because I would argue we should take care of it when we add the Site Title block (and maybe Site Description too).

#### Testing instructions

* Follow the [FSE readme](https://github.com/Automattic/wp-calypso/blob/e759e056360b72576e124989c4174afef78bc1db/apps/full-site-editing/README.md#local-development), and setup both the plugin and the theme.
* Enable the Blank Theme.
* Create some template parts, and assemble them into a template which also contains a Post Content slot.
* Assign this template to a **page**. Save it and view (or preview) it.
* Make sure it looks like it looks in the editor.
* Make sure it looks like a Twenty Nineteen site.
* Make sure nothing blew up in the process.

Fix #33160

### CODE REVIEW

**Comment by vindl** [comment]
I'm running into the following fatal when trying to view the page, regardless of whether it has a template set or not:

```
Fatal error: Uncaught Error: Call to undefined function wp_body_open() in /var/www/html/wp-content/themes/full-site-editing/header-blank.php:11 Stack trace: #0 /var/www/html/wp-includes/template.php(704): require_once() #1 /var/www/html/wp-includes/template.php(653): load_template('/var/www/html/w...', true) #2 /var/www/html/wp-includes/general-template.php(41): locate_template(Array, true) #3 /var/www/html/wp-content/themes/full-site-editing/page.php(3): get_header('blank') #4 /var/www/html/wp-includes/template-loader.php(77): include('/var/www/html/w...') #5 /var/www/html/wp-blog-header.php(19): require_once('/var/www/html/w...') #6 /var/www/html/index.php(17): require('/var/www/html/w...') #7 {main} thrown in /var/www/html/wp-content/themes/full-site-editing/header-blank.php on line 11
```

**Comment by Copons** [comment]
> I'm running into the following fatal when trying to view the page, regardless of whether it has a template set or not:
> 
> ```
> Fatal error: Uncaught Error: Call to undefined function wp_body_open() in /var/www/html/wp-content/themes/full-site-editing/header-blank.php:11 Stack trace: #0 /var/www/html/wp-includes/template.php(704): require_once() #1 /var/www/html/wp-includes/template.php(653): load_template('/var/www/html/w...', true) #2 /var/www/html/wp-includes/general-template.php(41): locate_template(Array, true) #3 /var/www/html/wp-content/themes/full-site-editing/page.php(3): get_header('blank') #4 /var/www/html/wp-includes/template-loader.php(77): include('/var/www/html/w...') #5 /var/www/html/wp-blog-header.php(19): require_once('/var/www/html/w...') #6 /var/www/html/index.php(17): require('/var/www/html/w...') #7 {main} thrown in /var/www/html/wp-content/themes/full-site-editing/header-blank.php on line 11
> ```

Somebody needs to update their WP 😏 

https://developer.wordpress.org/reference/functions/wp_body_open/

Jokes aside, @vindl: the function was introduced in 5.2.0.
I've seen it used in the [Twenty Nineteen header file](https://themes.trac.wordpress.org/browser/twentynineteen/1.4/header.php#L23), and just copied without giving it a second thought.

Turns out that Twenty Nineteen introduced it in v1.4 (so you might need to update the theme too! 😛), and it "emulates" it as a [fallback](https://themes.trac.wordpress.org/browser/twentynineteen/1.4/inc/template-tags.php#L242).

In other words, the function exists if:
- WP > 5.2.0
- Twenty Nineteen > 1.4

I presume they are both always true on WPCOM, but I'll add a `function_exists` check just to be 100% sure.

**Comment by Copons** [comment]
Update
In d37aa19 I've added some Twenty Nineteen styles to the Blank Theme, and enqueued its stylesheet as editor style.

This fixes a visual issue happening when Template Parts contains blocks using a dark background color, which automatically gives them a white text color.

The Twenty Nineteen editor style (which previously was the only one enqueued in the Blank Theme editor) only handles the text color but not the background color, which is added as inline style to the block content.
So, when adding these Template Parts to a template, they would be rendered as white text on white background.

The fix simply adds the background color rules into the `.template-block__content` class, which is only used inside the editor (when rendered in the front-end the class is `.template-part`).

**Comment by mattwiebe** [comment]
Gonna just go ahead and merge this since @copons is now out on leave

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/index.php [hunk 1]
```diff
@@ -15,14 +15,23 @@ function render_template_block( $attributes ) {
...
+	setup_postdata( $template );
+	ob_start();
+	?>
 
+		<div class="template-part<?php echo $align; ?>">
+			<?php echo apply_filters( 'the_content', get_the_content() );  ?>
+		</div><!-- .template-part -->
+
+	<?php
+	$content = ob_get_clean();
```
Comment (dmsnell) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/index.php]: what's the point of `ob_start()` here? is it to handle string interpolation as HTML vs. as a PHP string?

```php
setup_postdata( $template );
$content = sprintf(
	'<div class="template-part%s">%s</div><!-- .template-part -->',
	$align,
	apply_filters( 'the_content', get_the_content() )
);
wp_reset_postdata();
```

**Comment by dmsnell** [comment]
Do we have a guide to how this theme is working? It'd be helpful to have it documented from a high-level description with some diagrams and explanations of the data flow of the templates so that we don't have to find and digest the code to answer those basic questions. This is even more valuable considering that there's still uncertainty on whether this solution will pan out in the end.
